src/db/util/raw-model.ts

```diff
@@ -229,6 +263,25 @@ export const defineRawModel =
       await builder.truncate();
     };
 
+    const count: CountFn<F> = async ({ where, trx, distinct } = {}) => {
+      const builder = trx ? masterTable().transacting(trx) : replicaTable();
+      const query = builder.where(prepareCondition(where ?? {}));
+
+      if (distinct !== undefined) {
+        query.countDistinct(distinct);
+      } else {
+        query.count('*');
+      }
+
+      const res = await query;
+
+      const mappedResponse = res as Array<{ count: string }>;
+
+      return mappedResponse.length > 0
+        ? parseInt(mappedResponse[0].count, 10)
+        : 0;
```

Comment:
Since PostgreSQL returns `bigint` type, type which will be represented as a string, I would type the response even stronger, as a tuple (array with single element) that contains count object.

Next, I would not try to convert it back to a number, because there is a valid reason it is a string. So, let's convert to a `BigInt` and don't return 0 as a default value, especially if response is invalid (has zero elements in your case).

Let's rewrite like this
```ts
const res = (await query) as unknown as [{ count: string }];

if (res.length !== 1) {
  throw new Error('Count result must be an array of single element');
}

return BigInt(res[0].count);
```
or use `io-ts` to validate the response fully, not only that it has a single element
```ts
const COUNT = t.tuple([
  t.type({
    count: t.string,
  }),
]);

const decode = COUNT.decode(res);

if (!isRight(decode)) {
  const errors = ioTsErrorFormatter(decode);
  throw new DataValidationError({
    errors,
    identifier: 'count query',
    data: res,
  });
}

return BigInt(decode.right[0].count);
```
I think we should use the first, simpler option, because response doesn't really have much room to violate the shape.

Reply:
I think that the first option is the best - There is no need to add that level of validation for a simple query like this one.